schemes.go

```diff
@@ -921,6 +925,84 @@ func (scheme *SchemeManager) parseContents(conf *Configuration) error {
 		conf.Issuers[issuer.Identifier()] = issuer
 		return scheme.parseCredentialsFolder(conf, issuer, filepath.Join(dir, "Issues"))
 	})
+
+	if err != nil {
+		return err
+	}
+
+	// validate that there are no circular dependencies
+	for _, credType := range conf.CredentialTypes {
+		if credType.SchemeManagerID == scheme.ID {
+			if err := credType.validateDependencies(conf, []CredentialTypeIdentifier{}, credType.Identifier()); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func (ct CredentialType) validateDependencies(conf *Configuration, validatedDeps DependencyChain, toBeChecked CredentialTypeIdentifier) error {
+	if len(validatedDeps) >= maxDepComplexity {
+		return errors.New("dependency tree too complex: " + validatedDeps.String())
+	}
+	for _, outer := range conf.CredentialTypes[ct.Identifier()].Dependencies {
+		// at least one middle needs to be valid
+		atLeastOneInnerConSatisfied := false
```

Comment:
You might consider to use the same variable names for these booleans as `IssueWizardContents.Choose()`, which uses two booleans in exactly the same fashion - or rename those to these.